### PR TITLE
cleanup: small tweaks to CMake super build

### PR DESCRIPTION
**Pull request by coryan**
This moves the super build files from ci/super/ to super/ because we use
it in more than just the CI builds, and also because it is there for
other repositories.
I also cleaned up the instructions to install the dependencies on $HOME,
to make them look like the instructions in g-c-cpp-common.
Finally I removed a TODO that was no longer applicable.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/948)

### CODE REVIEW

File: doc/setup-cmake-environment.md [hunk 1]
```diff
@@ -39,32 +39,36 @@ The build takes a few minutes the first time, as it needs to create a Docker
...
-compile the project against these libraries. The installation needs to be done
-every time the version of the dependencies is changed, and it is not done
-automatically. But once installed you can use them for any build.
+You will install the dependencies in `$HOME/local-spanner` (or a similar
+directory), and compile the project against these libraries. The installation
+needs to be done every time the version of the dependencies is changed, and it
+is not done automatically. But once installed you can use them for any build.
 
-Configure the super-build to install in `$HOME/local`:
+Configure the super-build to install in `$HOME/local-spanner`, we recommend that
```
Comment (devbww) [inline doc/setup-cmake-environment.md]: s/, we/. We/

File: doc/setup-cmake-environment.md [hunk 2]
```diff
@@ -39,32 +39,36 @@ The build takes a few minutes the first time, as it needs to create a Docker
...
+use use Ninja for this build because it is substantially faster than Make in
```
Comment (devbww) [inline doc/setup-cmake-environment.md]: s/use use/you use/

File: doc/setup-cmake-environment.md (hunk 1 above)
Comment (coryan) [reply doc/setup-cmake-environment.md]: Done, thanks!

File: doc/setup-cmake-environment.md (hunk 2 above)
Comment (coryan) [reply doc/setup-cmake-environment.md]: Done, thanks!

**Pull request merged**